src/Result.php

```diff
@@ -4,22 +4,25 @@
 namespace Yiisoft\Validator;
 
 /**
- * Result represents validation result
+ * Result represents a single value validation result.
+ * It may either be success or contain one or multiple errors.
  */
 final class Result
 {
-    private $valid = true;
-    private $errors = [];
+    private bool $valid = true;
+    private array $errors = [];
 
     public function isValid(): bool
     {
         return $this->valid;
     }
 
-    public function addError(string $message): void
+    public function addError(string $message): self
```

Comment:
Why it made like a fluent method? Class `Result` is just storage for error messages, not a builder.
I think you need to revert your changes.

Reply:
Yes. You're right.